### ISSUE TITLE
Adding BlockProducerRegistration utilities to `sidechain-main-cli`

### ISSUE THREAD

**Issue by jaredponn** [input-output-hk/partner-chains-smart-contracts#23]
## Description

In input-output-hk/trustless-sidechain#566, almost all functionalities of `trustless-sidechain-gen-signatures` was superseded by `sidechain-main-cli`.

One functionality that was missing was utilities for BlockProducerRegistration.
This issue requests that we fill this in + finally deprecate `trustless-sidechain-gen-signatures`

## Blocked By
- input-output-hk/trustless-sidechain#566 [or one can build on top of it]

## Goals
- Update the docs to show  how block producer registration can be done.
